src/common/mod.rs

```diff
@@ -112,15 +110,41 @@ macro_rules! heatmap {
             name = $name,
             crate = metriken
         )]
-        pub static $ident: metriken::Heatmap = metriken::Heatmap::new(0, 8, 64, core::time::Duration::from_secs(1), core::time::Duration::from_millis(100));
+        pub static $ident: metriken::AtomicHistogram = metriken::AtomicHistogram::new(7, 64);
+    };
+    ($ident:ident, $name:tt, $description:tt) => {
+        #[metriken::metric(
+            name = $name,
+            description = $description,
+            crate = metriken
+        )]
+        pub static $ident: metriken::AtomicHistogram = metriken::AtomicHistogram::new(7, 64);
+    };
+}
+
+#[macro_export]
+#[rustfmt::skip]
+/// A convenience macro for constructing a lazily initialized
+/// `metriken::Heatmap` given an identifier, name, and optional description.
+///
+/// The heatmap configuration used here can record counts for all 64bit integer
+/// values with a maximum error of 0.78%. The heatmap covers a moving window of
+/// one minute with one second resolution.
+macro_rules! bpfhistogram {
+    ($ident:ident, $name:tt) => {
+        #[metriken::metric(
+            name = $name,
+            crate = metriken
+        )]
+        pub static $ident: metriken::RwLockHistogram = metriken::RwLockHistogram::new(7, 64);
```

Comment:
No strong opinion, just as a thought: is it worth making 7 and 64 consts so that it is easier to change them if we decide we want to track at a different granularity (7 being more likely to change).

Reply:
This is not really possible to change without other code changes, the indexing in the BPF progs and in Rezolus need to agree exactly. We don't have a mechanism to replace the indexing function in the BPF prog if we change the histogram params.

Reply:
Ack, makes sense, thanks.